cmd/krel/cmd/release_notes.go

```diff
@@ -206,27 +276,83 @@ func runReleaseNotes() (err error) {
 	}
 
-	// TODO: implement PR creation for k-sigs/release-notes
 	return nil
 }
 
 // validateDraftPROptions checks if we have all needed parameters to create the Release Notes PR
 func validateDraftPROptions() error {
 	if releaseNotesOpts.createDraftPR {
+		// Check if both --create-website-pr --create-draft-pr --dra are set
+		if releaseNotesOpts.createWebsitePR {
+			return errors.New("Cannot create release notes draft if --create-website-pr is set")
+		}
+
 		// Check if --draft-org is set
 		if releaseNotesOpts.draftOrg == "" {
-			logrus.Warn("cannot generate the Release Notes draft PR without --draft-org")
+			logrus.Warn("cannot generate the Release Notes PR without --draft-org")
 		}
 
 		// Check if --draft-repo is set
 		if releaseNotesOpts.draftRepo == "" {
-			logrus.Warn("cannot generate the Release Notes draft PR without --draft-repo")
+			logrus.Warn("cannot generate the Release Notes PR without --draft-repo")
 		}
 
 		if releaseNotesOpts.draftOrg == "" || releaseNotesOpts.draftRepo == "" {
-			return errors.New("To generate the release notes draft you must define both --draft-org and --draft-repo")
+			return errors.New("To generate the release notes PR you must define both --draft-org and --draft-repo")
 		}
 	}
+
+	return nil
+}
+
+// validateWebsitePROptions checks if we have all needed parameters to create the Release Notes PR
+func validateWebsitePROptions() error {
+	if releaseNotesOpts.createWebsitePR {
+		// Check if --website-org is set
+		if releaseNotesOpts.websiteOrg == "" {
+			logrus.Warn("cannot generate the Website PR without --website-org")
+		}
+
+		// Check if --website-repo is set
+		if releaseNotesOpts.websiteRepo == "" {
+			logrus.Warn("cannot generate the Website PR without --website-repo")
+		}
+
+		if releaseNotesOpts.websiteOrg == "" || releaseNotesOpts.websiteRepo == "" {
+			return errors.New("To generate the website PR you must define both --website-org and --website-repo")
+		}
+	}
+	return nil
+}
+
+// PushToRemote gets a repo and pushes its changes to userForkName
+func pushToRemote(repository *git.Repo, branchname string) error {
+	// amend the latest commit
+	userName, err := command.New("git", "config", "--get", "user.name").RunSuccessOutput()
+	if err != nil {
+		return errors.Wrap(err, "while trying to get the user's name")
+	}
+
+	userEmail, err := command.New("git", "config", "--get", "user.email").RunSuccessOutput()
+	if err != nil {
+		return errors.Wrap(err, "while trying to get the user's name")
+	}
+
+	logrus.Infof("amending the last commit to the user's config: %v <%v>", userName.OutputTrimNL(), userEmail.OutputTrimNL())
+
+	if err := command.NewWithWorkDir(
+		repository.Dir(), "git", "commit", "--amend", "--signoff", "--no-edit",
+		"--author", fmt.Sprintf("%v <%v>", userName.OutputTrimNL(), userEmail.OutputTrimNL()),
+	).RunSilentSuccess(); err != nil {
+		return errors.Wrap(err, "amending commit with user's data")
+	}
```

Comment:
I feel that we should change the `repo.Commit` method to streamline this. Can we add a TODO here to indicate that this might be simplified?

Reply:
I've added a new Commit function `CommitWithOptions` to the git package which can get custom options for the commit. I modified the `Commit` function to use it and to send the release notes PRs on behalf of the user, I added a new `UserCommit` function which reads the user data from the local git.